src/data/client.rs

```diff
@@ -0,0 +1,313 @@
+            cache_duration: Duration::minutes(5), // Cache for 5 minutes
+        }
+    }
+
+    pub async fn get_all_stations(&mut self, include_real_time: bool) -> Result<Vec<VelibStation>> {
+        let reference_stations = self.get_reference_stations().await?;
+
+        if !include_real_time {
+            return Ok(reference_stations
+                .into_values()
+                .map(VelibStation::new)
+                .collect());
+        }
+
+        let real_time_data = self.get_realtime_data().await?;
+
+        let stations = reference_stations
+            .into_iter()
+            .map(|(code, reference)| {
+                let mut station = VelibStation::new(reference);
+                if let Some(real_time) = real_time_data.get(&code) {
+                    station = station.with_real_time(real_time.clone());
+                }
+                station
+            })
+            .collect();
+
+        Ok(stations)
+    }
+
+    pub async fn get_station_by_code(
+        &mut self,
+        station_code: &str,
+        include_real_time: bool,
+    ) -> Result<Option<VelibStation>> {
+        let reference_stations = self.get_reference_stations().await?;
+
+        if let Some(reference) = reference_stations.get(station_code) {
+            let mut station = VelibStation::new(reference.clone());
+
+            if include_real_time {
+                let real_time_data = self.get_realtime_data().await?;
+                if let Some(real_time) = real_time_data.get(station_code) {
+                    station = station.with_real_time(real_time.clone());
+                }
+            }
+
+            Ok(Some(station))
+        } else {
+            Ok(None)
+        }
+    }
+
+    async fn get_reference_stations(&mut self) -> Result<HashMap<String, StationReference>> {
+        // Check cache
+        if let Some((ref cached_data, cached_time)) = &self.stations_cache {
+            if Utc::now().signed_duration_since(*cached_time) < self.cache_duration {
+                return Ok(cached_data.clone());
+            }
+        }
+
+        info!("Fetching reference stations from API");
+
+        // Fetch data with pagination (API limit is 100 records per request)
+        let mut stations = HashMap::new();
+        let mut offset = 0;
+        let limit = 100;
+
+        loop {
+            let response = self
+                .client
+                .get(VELIB_STATIONS_URL)
+                .query(&[
+                    ("limit", &limit.to_string()),
+                    ("offset", &offset.to_string()),
+                ])
+                .send()
+                .await?;
+
+            if !response.status().is_success() {
+                let status = response.status();
+                let body = response
+                    .text()
+                    .await
+                    .unwrap_or_else(|_| "No body".to_string());
+                return Err(Error::Internal(anyhow::anyhow!(
```

Comment:
Do we need some kind of exponential backoff? It seems odd to fail fast on a paginated API like this.